include/picongpu/plugins/radiation/Radiation.hpp

```diff
@@ -664,396 +659,197 @@ namespace picongpu
-                    hdf5DataFile.writeGlobalAttribute(
-                        ctIterationEncoding,
-                        "iterationEncoding",
-                        iterationEncoding.c_str());
-
-                    /* the ..._0_0_0... extension comes from the current filename
-                       formating of the serial data colector in libSplash */
-                    const int indexCutDirectory = name.rfind('/');
-                    std::string iterationFormat(name.substr(indexCutDirectory + 1) + std::string("%T_0_0_0.h5"));
-                    splash::ColTypeString ctIterationFormat(iterationFormat.length());
-                    hdf5DataFile.writeGlobalAttribute(ctIterationFormat, "iterationFormat", iterationFormat.c_str());
-
-                    hdf5DataFile.writeAttribute(currentStep, splashFloatXType, nullptr, "dt", &DELTA_T);
+                    openPMDdataFileIteration.setDt<float>(DELTA_T);
                     const float_X time = float_X(currentStep) * DELTA_T;
-                    hdf5DataFile.writeAttribute(currentStep, splashFloatXType, nullptr, "time", &time);
-                    splash::ColTypeDouble ctDouble;
-                    hdf5DataFile.writeAttribute(currentStep, ctDouble, nullptr, "timeUnitSI", &UNIT_TIME);
-
+                    openPMDdataFileIteration.setTime(time);
+                    openPMDdataFileIteration.setTimeUnitSI(UNIT_TIME);
                     /* end required openPMD global attributes */
 
                     /* begin recommended openPMD global attributes */
-
-                    std::string author = Environment<>::get().SimulationDescription().getAuthor();
-                    if(author.length() > 0)
-                    {
-                        splash::ColTypeString ctAuthor(author.length());
-                        hdf5DataFile.writeGlobalAttribute(ctAuthor, "author", author.c_str());
-                    }
-
-                    std::string software("PIConGPU");
-                    splash::ColTypeString ctSoftware(software.length());
-                    hdf5DataFile.writeGlobalAttribute(ctSoftware, "software", software.c_str());
-
+                    const std::string software("PIConGPU");
                     std::stringstream softwareVersion;
                     softwareVersion << PICONGPU_VERSION_MAJOR << "." << PICONGPU_VERSION_MINOR << "."
                                     << PICONGPU_VERSION_PATCH;
                     if(!std::string(PICONGPU_VERSION_LABEL).empty())
                         softwareVersion << "-" << PICONGPU_VERSION_LABEL;
-                    splash::ColTypeString ctSoftwareVersion(softwareVersion.str().length());
-                    hdf5DataFile.writeGlobalAttribute(
-                        ctSoftwareVersion,
-                        "softwareVersion",
-                        softwareVersion.str().c_str());
+                    openPMDdataFile.setSoftware(software, softwareVersion.str());
```

Comment:
```suggestion
                    openPMDdataFile.setSoftware(software, softwareVersion);
```

Reply:
Ah sorry, that suggestion was wrong. I was thinking of `c_str()` when reading that, the original version was correct.